src/libtcod/renderer_xterm.c

```diff
@@ -159,7 +174,125 @@ static void xterm_destructor(struct TCOD_Context* __restrict self) {
   xterm_cleanup();
 }
 
-TCOD_Context* TCOD_renderer_init_xterm(const char* window_title) {
+static void xterm_handle_input_escape() {
+  if (getchar() != '[')
+    return;
+  char rows_buf[16], cols_buf[16];
+  rows_buf[15] = '\0';
+  cols_buf[15] = '\0';
```

Comment:
These character buffers should be zero initialized to avoid any kind of possible issues later.
```suggestion
  char rows_buf[16] = "";
  char cols_buf[16] = "";
```